bids/layout/layout.py

```diff
@@ -673,10 +674,13 @@ def get(self, return_type='object', target=None, scope='all',
                                  'target entity must also be specified.')
 
             if return_type == 'id':
-                ent_iter = (x.entities for x in results)
-                results = list({
-                    ents[target] for ents in ent_iter if target in ents
-                })
+                u_results = set()
+                for f in results:
+                    if target in f.entities:
+                        val = f.entities[target]
+                        if isinstance(val, Hashable):
+                            u_results.add(val)
+                results = list(u_results)
```

Comment:
This really just needs to add `if isinstance()` to the conditional in the comprehension:

```suggestion
                results = list(dict.from_keys(
                    ents[target] for ents in ent_iter
                    if target in ents and isinstance(ents[target], Hashable)
                ))
```

I also switched to `dict.from_keys()` instead of `set()` because it will preserve insertion order, so it acts like a lightweight `OrderedSet()`.

Reply:
Yes, I wasn't sure if accessing `f.entities[target]` twice (rather than saving in variable `var`) had any speed implications.

Reply:
This looks fine though. They're probably minimal